browser/brave_ads/brave_ads_driver_delegate_impl.cc

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/brave_ads_driver_delegate_impl.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/extensions/api/brave_action_api.h"
+#include "brave/browser/extensions/brave_component_loader.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "chrome/browser/extensions/extension_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "content/public/browser/web_contents.h"
+#include "extensions/browser/extension_system.h"
+#include "extensions/common/constants.h"
+
+namespace brave_ads {
+
+namespace {
+
+constexpr char kAdsEnableRelativeUrl[] =
+    "request_ads_enabled_panel.html#enable_ads";
+
+}  // namespace
+
+BraveAdsDriverDelegateImpl::BraveAdsDriverDelegateImpl(
+    Profile* profile,
+    content::WebContents* web_contents,
+    brave_rewards::RewardsService* rewards_service)
+    : WebContentsObserver(web_contents),
+      profile_(profile),
+      web_contents_(web_contents),
+      rewards_service_(rewards_service) {
+  DCHECK(profile_);
+  DCHECK(web_contents_);
+}
+
+BraveAdsDriverDelegateImpl::~BraveAdsDriverDelegateImpl() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  DCHECK(!callback_);
+}
+
+void BraveAdsDriverDelegateImpl::RequestAdsEnabled(
+    RequestAdsEnabledCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  if (!rewards_service_ || callback_) {
+    std::move(callback).Run(false);
+    return;
+  }
+  callback_ = std::move(callback);
+
+  rewards_service_->StartProcess(base::DoNothing());
```

Comment:
we should not be starting rewards here, especially with no callback. We should be enabling rewards and then rewards will start on its own.

Reply:
you can see here that rewards is started automatically if ads or auto contribute are enabled https://github.com/brave/brave-core/blob/e0d0345187b9387f8a195351bacddb08e40cde08/components/brave_rewards/browser/rewards_service_impl.cc#L413

Reply:
Fixed